File: backend/entity/StudentSkill.go
package entity

import (
	"gorm.io/gorm"
)

type StudentSkill struct {
	gorm.Model
	SkillID   uint  `json:"skill_id"`
	StudentID uint  `json:"student_id"`
	Skill     Skill `gorm:"foreignKey:SkillID"`
	Student   Student `gorm:"foreignKey:StudentID"`
}

File: frontend/src/interfaces/StudentSkill.ts
import type { SkillInterface } from './Skill';
import type { StudentInterface } from './Student';

export interface StudentSkillInterface {
  id?: number;  
  ID?: number;          // จาก gorm.Model.ID
  SkillID: number;
  StudentID: number;
  Skill?: SkillInterface;      // ความสัมพันธ์ (optional)
  Student?: StudentInterface;  // ความสัมพันธ์ (optional)
  CreatedAt?: string;
  UpdatedAt?: string;
  DeletedAt?: string | null;
}